app/controllers/spree/braintree_client_token_controller.rb

```diff
@@ -0,0 +1,38 @@
+module Spree
+  class BraintreeClientTokenController < Spree::BaseController
+    protect_from_forgery except: [:create]
+
+    def create
+      respond_to do |format|
+        format.json do
+          render json: {
+            client_token: generate_client_token,
+            payment_method_id: gateway.id
+          }
+        end
+      end
+    end
+
+    private
+
+    def gateway
+      @gateway ||= if params[:payment_method_id]
+        Solidus::Gateway::BraintreeGateway.find_by!(id: params[:payment_method_id])
+      else
+        Solidus::Gateway::BraintreeGateway.find_by!(active: true)
+      end
+    end
+
+    def generate_client_token
+      options = {}
+      options[:customer_id] = customer_id if customer_id.present?
+
+      gateway.generate_client_token(options)
+    end
+
+    def customer_id
```

Comment:
This is sufficient most the time but if the current user is an admin, then the new payment form is built on the admin's customer id and not the order's user.